apps/sms/services/js/drafts.js

```diff
@@ -46,11 +46,7 @@
       drafts.push(draft);
       draftIndex.set(draft.threadId, drafts);
 
-      this.store();
-
-      this.emit('saved', draft);
-
-      return this;
+      return this.store().then(() => this.emit('saved', draft));
```

Comment:
yes why not.

We have actually 2 different cases here.
1. the JS object is modified and so other parts of the app can already use it
2. the backend has been saved

That's why I think it makes sense to have the 2 events (and possibly promises). Basically some parts of the code will need to wait only 1, but some other parts will need to wait until 2.